### PR TITLE
Handle if reading tsconfig file fails

### LINKED ISSUE #36862: applyChangesInOpenFiles crashes for file in MacOS trash
From https://github.com/microsoft/vscode/issues/90769

**TypeScript Version:** 3.9.0-dev.20200215

**Search Terms:**
- updateOpen

**Repo**
1. In MacOS
1. Move a js file to the trash.
1. Now open that file in VS Code

**Bug**

```
Trace  - 19:35:32.409] <semantic> Response received: updateOpen (2). Request took 1456 ms. Success: false . Message: Error processing request. Cannot read property 'length' of undefined
TypeError: Cannot read property 'length' of undefined
    at createSourceFile (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:19142:110)
    at Object.parseJsonText (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:18966:26)
    at Object.parseJsonText (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:18787:23)
    at ProjectService.loadConfiguredProject (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:142182:33)
    at ProjectService.createAndLoadConfiguredProject (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:142164:22)
    at ProjectService.createLoadAndUpdateConfiguredProject (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:142169:36)
    at ProjectService.assignProjectToOpenedScriptInfo (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:143028:44)
    at /Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:143324:107
    at Object.mapDefined (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:562:30)
    at ProjectService.applyChangesInOpenFiles (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:143324:48)
    at Session.handlers.ts.createMapFromTemplate._a.<computed> (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:144188:46)
    at /Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:145931:88
    at IOSession.Session.executeWithRequestId (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:145922:28)
    at IOSession.Session.executeCommand (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:145931:33)
    at IOSession.Session.onMessage (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:145955:35)
    at Interface.<anonymous> (/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js:147270:27)
    at Interface.emit (events.js:203:13)
    at Interface._onLine (readline.js:316:10)
    at Interface._normalWrite (readline.js:461:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:203:13)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:276:11)
    at Socket.Readable.push (_stream_readable.js:210:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)
```

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes #36862

### CODE REVIEW

File: src/server/editorServices.ts [hunk 1]
```diff
@@ -1974,7 +1974,7 @@ namespace ts.server {
             // Read updated contents from disk
             const configFilename = normalizePath(project.getConfigFilePath());
 
-            const configFileContent = this.host.readFile(configFilename)!; // TODO: GH#18217
+            const configFileContent = this.host.readFile(configFilename) || "";
```
Comment (amcasey) [inline src/server/editorServices.ts]: I usually think of `{}` as the simplest tsconfig.  What does `parseJsonText` return when it's empty?  I some sort of diagnostic produced?

Comment (sheetalkamat) [reply src/server/editorServices.ts]: It will create source file with no statements and parseJsonSourceFileConfigFileContent will take that to mean `{}`

Comment (amcasey) [reply src/server/editorServices.ts]: Cool.  Will some sort of diagnostic be produced for the failed read?

Comment (sheetalkamat) [reply src/server/editorServices.ts]: created #37611

**Pull request merged**